### PR TITLE
[CELEBORN-983] Rename PrometheusMetric configuration

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
Replace
```properties
celeborn.metrics.master.prometheus.host
celeborn.metrics.master.prometheus.port
celeborn.metrics.worker.prometheus.host
celeborn.metrics.worker.prometheus.port
```
With
```properties
celeborn.master.http.host
celeborn.master.http.port
celeborn.worker.http.host
celeborn.worker.http.port
```

### Why are the changes needed?
The `celeborn.master.metrics.prometheus.port` and `celeborn.metrics.worker.prometheus.port` bind port not only serve prometheus metrics, but also provide some useful API services.

https://celeborn.apache.org/docs/latest/monitoring/#rest-api

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1595,6 +1607,17 @@ object CelebornConf extends Logging {
       .checkValue(p => p >= 1024 && p < 65535, "Invalid port")
       .createWithDefault(9097)
 
+  val MASTER_HTTP_SERVICE_PORT: ConfigEntry[Int] =
+    buildConf("celeborn.master.http.service.port")
```
Comment (jiaoqingbo) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1611]: udpate the migration.md？

File: worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala [hunk 2]
```diff
@@ -52,7 +52,7 @@ trait MiniClusterFeature extends Logging {
     val conf = new CelebornConf()
     conf.set(CelebornConf.METRICS_ENABLED.key, "false")
     val prometheusPort = masterPrometheusPort.getAndIncrement()
-    conf.set(CelebornConf.MASTER_PROMETHEUS_PORT.key, s"$prometheusPort")
+    conf.set(CelebornConf.MASTER_HTTP_SERVICE_PORT.key, s"$prometheusPort")
     logInfo(s"set celeborn.master.metrics.prometheus.port to $prometheusPort")
     if (map != null) {
```
Comment (jiaoqingbo) [inline worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala:57]: the log message should be changed，at the same time, the variable names masterPrometheusPort and workerPrometheusPort should remain consistent with the configuration items.

**Review by jiaoqingbo — APPROVED** [review]

**Comment by SteNicholas** [comment]
@cxzl25, I don't think there is a need to configure an override, otherwise there will be compatibility issues.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -2074,6 +2097,27 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("1000ms")
 
+  val WORKER_HTTP_SERVICE_HOST: ConfigEntry[String] =
+    buildConf("celeborn.worker.http.service.host")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2101]: the namespace "service" seems redundant

File: docs/monitoring.md [hunk 4]
```diff
@@ -269,17 +269,17 @@ an easy way to create new visualizations and monitoring tools for Celeborn and
 also easy for users to get the running status of the service. The REST API is available for
 both master and worker. The endpoints are mounted at `host:port`. For example,
 for the master, they would typically be accessible
-at `http://<master-prometheus-host>:<master-prometheus-port><path>`, and
-for the worker, at `http://<worker-prometheus-host>:<worker-prometheus-port><path>`.
+at `http://<master-http-service-host>:<master-http-service-port><path>`, and
+for the worker, at `http://<worker-http-service-host>:<worker-http-service-port><path>`.
```
Comment (cfmcgrady) [inline docs/monitoring.md:273]:
```suggestion
at `http://<master-http-host>:<master-http-port><path>`, and
for the worker, at `http://<worker-http-host>:<worker-http-port><path>`.
```

File: docs/monitoring.md [hunk 5]
```diff
@@ -269,17 +269,17 @@ an easy way to create new visualizations and monitoring tools for Celeborn and
...
 
-The configuration of `<master-prometheus-host>`, `<master-prometheus-port>`, `<worker-prometheus-host>`, `<worker-prometheus-port>` as below:
+The configuration of `<master-http-service-host>`, `<master-http-service-port>`, `<worker-http-service-host>`, `<worker-http-service-port>` as below:
```
Comment (cfmcgrady) [inline docs/monitoring.md:275]:
```suggestion
The configuration of `<master-http-host>`, `<master-http-port>`, `<worker-http-host>`, `<worker-http-port>` as below:
```

**Comment by cxzl25** [comment]
> I don't think there is a need to replace these configurations, otherwise there will be compatibility issues.

`celeborn.master.http.port` has automatic fallback of the original parameters `celeborn.metrics.master.prometheus.port`.

The current parameter is more ambiguous. It is not just a port opened for prometheus.

**Review by SteNicholas — APPROVED** [review]
LGTM.

File: docs/migration.md [hunk 6]
```diff
@@ -28,6 +28,16 @@ license: |
 
 - Since 0.4.0, Celeborn won't support `org.apache.spark.shuffle.celeborn.RssShuffleManager`.
 
+- - Since 0.4.0, Celeborn has changed the names of the prometheus master and worker configuration as shown in the table below:
```
Comment (AngersZhuuuu) [inline docs/migration.md:31]: Only one `-`

File: worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala [hunk 7]
```diff
@@ -72,7 +72,9 @@ trait MiniClusterFeature extends Logging {
     conf.set(CelebornConf.WORKER_STORAGE_DIRS.key, createTmpDir())
     conf.set(CelebornConf.WORKER_DISK_MONITOR_ENABLED.key, "false")
     conf.set(CelebornConf.CLIENT_PUSH_BUFFER_MAX_SIZE.key, "256K")
-    conf.set(CelebornConf.WORKER_PROMETHEUS_PORT.key, s"${workerPrometheusPort.incrementAndGet()}")
+    conf.set(
+      CelebornConf.WORKER_HTTP_PORT.key,
+      s"${workerHttpPort.incrementAndGet()}")
```
Comment (AngersZhuuuu) [inline worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala:77]: One line should be ok?

File: docs/monitoring.md [hunk 8]
```diff
@@ -269,17 +269,17 @@ an easy way to create new visualizations and monitoring tools for Celeborn and
...
-| celeborn.metrics.master.prometheus.host | 0.0.0.0 | Master's Prometheus host.  | 0.2.0 |
-| celeborn.metrics.master.prometheus.port | 9098    | Master's Prometheus port.  | 0.2.0 |
-| celeborn.metrics.worker.prometheus.host | 0.0.0.0 | Worker's Prometheus host.  | 0.2.0 |
-| celeborn.metrics.worker.prometheus.port | 9096    | Worker's Prometheus port.  | 0.2.0 |
+| Key                       | Default | Description         | Since |
+|---------------------------|---------|---------------------|-------|
+| celeborn.master.http.host | 0.0.0.0 | Master's http host. | 0.4.0 |
+| celeborn.master.http.port | 9098    | Master's http port. | 0.4.0 |
+| celeborn.worker.http.host | 0.0.0.0 | Worker's http host. | 0.4.0 |
+| celeborn.worker.http.port | 9096    | Worker's http port. | 0.4.0 |
```
Comment (AngersZhuuuu) [inline docs/monitoring.md:282]: Maybe we'd better to add a section about how to connect prometheus ?

File: docs/migration.md [hunk 9]
```diff
@@ -30,6 +30,19 @@ license: |
...
+    | Key Before v0.4.0                         | Key After v0.4.0            |
+    |-------------------------------------------|-----------------------------|
+    | `celeborn.metrics.master.prometheus.host` | `celeborn.master.http.host` |
+    | `celeborn.metrics.master.prometheus.port` | `celeborn.master.http.port` |
+    | `celeborn.metrics.worker.prometheus.host` | `celeborn.worker.http.host` |
+    | `celeborn.metrics.worker.prometheus.port` | `celeborn.worker.http.port` |
+
+## Upgrading from 0.3.1 to 0.3.2
+
+- Since 0.3.2, Celeborn changed the default value of `celeborn.worker.monitor.disk.check.interval` from `60` to `30`.
```
Comment (AngersZhuuuu) [inline docs/migration.md:44]: Remove this?

File: docs/monitoring.md (hunk 8 above)
Comment (cxzl25) [reply docs/monitoring.md:282]: There are related descriptions in the API list.

https://github.com/apache/incubator-celeborn/blob/56276e910f3489bc070846a19813e322b590fd17/docs/monitoring.md?plain=1#L292

**Review by waitinfuture — APPROVED** [review]
LGTM, Thanks! This change is more reasonable.

**Review by waitinfuture — APPROVED** [review]
LGTM, Thanks! This change is more reasonable.

**Pull request closed without merging**